### ISSUE TITLE
New-F5 Sessions throws confusing errors when login fails

### ISSUE THREAD

**Issue by kleinsmk** [joel74/POSH-LTM-Rest#166]
Lines 67 - 74 in New-F5session throws errors similar to below when a bad login is presented as well as line 39 :

ConvertFrom-Json : Cannot bind argument to parameter 'InputObject' because it is null.
At C:\Program Files\WindowsPowerShell\Modules\f5-ltm\1.4.239\Private\Invoke-F5RestMethod.ps1:39 char:50
+ ...         $message = $_.ErrorDetails.Message | ConvertFrom-json | Selec ...
+                                                  ~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [ConvertFrom-Json], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.ConvertFromJsonCommand
 
Invoke-F5RestMethod : "401 F5 Authorization Required: 
At C:\Program Files\WindowsPowerShell\Modules\f5-ltm\1.4.239\Public\New-F5Session.ps1:95 char:13
+     $JSON = Invoke-F5RestMethod -Method Get -Uri $VersionURL -F5Sessi ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Invoke-F5RestMethod
 
The Following code is responsible for the second set of errors as it assumes that the LTM name must be the reason for a failed webrequest.  The problem is a bad login will also throw the same unauthorized error.  I'd assume you could simply add and else and break but it looks like a lot more is going on here.

Try {
            Invoke-WebRequest -Uri $BaseURL -ErrorVariable LTMError -TimeoutSec 3
        }
        Catch {
            #If an error is thrown and it doesn't contain the word 'Unauthorized' then the LTM name and $BaseURL are invalid
            If ($LTMError[0] -notmatch 'Unauthorized'){
                Throw ("The specified LTM name $LTMName is not valid.")
            }
        }

**Comment by kleinsmk** [comment]
I apologize for the formatting.  I'm kinda git hub newbie.

**Comment by joel74** [comment]
Hi, I understand it's confusing but a request with incorrect credentials actual throws the error:
"ConvertFrom-Json : Cannot bind argument to parameter 'InputObject' because it is null."
It doesn't throw an 'Unauthorized' error. 

The result we're getting back from the LTM device when passing incorrect creds isn't in JSON format, so our attempt to convert to JSON fails. I've made a change and a[ pull request](https://github.com/joel74/POSH-LTM-Rest/pull/167) to the project so, if we don't get back an error in JSON format, we just treat it as a string and return that. I tested with incorrect creds and got a "401 F5 Authorization Required" returned.

I'll leave the pull request there until the end of the week for review and comments, and then merge it in.
